Tweak listviews

hawkrives opened the pull request:
This adds refreshing support to all of our listviews.

It also fixes the blank-screen bug in buildingHours, I believe.

Squash and merge this one, please, rather than just merging.

hawkrives committed:
add a refresh control to the news views

hawkrives committed:
check for offline in sis stuff

hawkrives committed:
add pageSize to newsContainer

hawkrives committed:
add a refreshControl to Calendar

hawkrives committed:
fix bug about getConnected in financials data loading

hawkrives committed:
make buildingHours update without needing to refresh

also fixes the missing content bug

hawkrives committed:
simplify refresh in calendar and news

hawkrives committed:
add refresh to oleville

hawkrives committed:
add refresh to balances

hawkrives committed:
add time delay to refresh in courses

drewvolz commented on views/building-hours/index.js:
```diff
@@ -54,6 +75,13 @@ export default class BuildingHoursView extends React.Component {
         renderRow={this._renderRow.bind(this)}
         pageSize={4}
         initialListSize={6}
+        removeClippedSubviews={false}  // remove after https://github.com/facebook/react-native/issues/8607#issuecomment-241715202
```
_Not an issue but a suggestion for later_

We will want to add this to every list view temporarily, perhaps? It has happened on more than one list view (Important Contacts for example). 

Come to think of it, we should implement our own wrapper around ListView to handle our customization. We would need a `prop` to tell if this were to exist inside of a Tabbar or not in the case of the extra bottom inset, but it should be doable.

That can come in a later PR. I am okay with this for now. Nice find!

hawkrives replied:
Can you make a list of all the listviews that you've seen this on? Are there any you haven't?

I should add this prop to all the affected ones now, as this PR.

drewvolz committed:
Fix function called with loading spinners.

This fixes the animations on the listviews when loading with activity spinners.
